### DISCUSSION TITLE
Configuring for version 1 hardware?

### DISCUSSION THREAD

**Discussion by JollyWizard** [GuruSR/Watchy_GSR#18] [Q&A] [answered]
I have an earlier watchy, v1.0, and had trouble with the button layouts not being the same on newer compilation tools.  

You can see the differences listed in `sqmfi/watchy/config.h`.  

There is some sort of collision between versions and another board signal, that makes the up button constantly trigger unless configured properly.

What is the best way to trigger the proper button config in your framework?

**Comment by GuruSR** [comment, marked as answer]
If you use Watchy GSR, it auto-configures itself on boot to whichever version you have, so the watchy/config.h is not in use and doesn't affect it at all.  You're totally safe to use this without having to worry about what version you have.  I also have the V1.0.

**Reply by JollyWizard** [reply]
That's great to hear. 

Based on your feedback, I dug deeper and found the relevant code in SmallRTC.  

Seems a little complicated.  How did you determine this method?

**Reply by GuruSR** [reply]
Most of the code in SmallRTC for figuring out the RTCs is from Watchy's original code (before it was broken out into WatchyRTC), the rest is testing code I derived to determine where the battery is as it moves on different versions, so as long as the voltage was there, it could tell what version it was, on the 1.5 and 2.0, if the battery wasn't, it'd wind up choosing 1.5, but most people don't use it that way, they use it as, well, a watch!

**Reply by JollyWizard** [reply]
Very interesting. 

Do you think that a 1.5 default is sensible in that scenario?  Do we know if the distribution of units would make 2.0 more likely?

**Reply by GuruSR** [reply]
1.5 probably is the simplest as the RTC is being seen, but if the battery test alone (with SmallRTC 2.3 now up and in use), it'll find any of the 3 versions based on the battery alone (in the event the external RTC is dead and not responding to anything).
